impl/game_common_lib/network_helpers.cpp

```diff
@@ -7,6 +7,9 @@
 void NetworkHelpers::freeSendString(
     std::string const& str, asio::ip::tcp::socket& socket, jt::LoggerInterface& logger)
 {
+    if (!socket.is_open()) {
```

Comment:
Just added some safety checks in the free helper functions to avoid reading/writing from closed socket.